packages/git/src/browser/git-widget.ts

```diff
@@ -12,16 +12,31 @@ import { GitWatcher, GitStatusChangeEvent } from '../common/git-watcher';
 import { GIT_RESOURCE_SCHEME } from './git-resource';
 import { MessageService, ResourceProvider, CommandService, DisposableCollection } from '@theia/core';
 import URI from '@theia/core/lib/common/uri';
-import { VirtualRenderer, ContextMenuRenderer, OpenerService, open } from '@theia/core/lib/browser';
+import { VirtualRenderer, ContextMenuRenderer, OpenerService, open, VirtualWidget, LabelProvider } from '@theia/core/lib/browser';
 import { h } from '@phosphor/virtualdom/lib';
 import { Message } from '@phosphor/messaging';
 import { DiffUris } from '@theia/editor/lib/browser/diff-uris';
 import { WorkspaceCommands } from '@theia/workspace/lib/browser/workspace-commands';
 import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
-import { GitBaseWidget, GitFileChangeNode } from './git-base-widget';
+import { GitRepositoryProvider } from './git-repository-provider';
+
+export interface GitFileChangeNode extends GitFileChange {
+    readonly icon: string;
+    readonly label: string;
+    readonly description: string;
+    readonly caption?: string;
+    readonly extraIconClassName?: string;
+    readonly commitSha?: string;
+    selected?: boolean;
+}
+export namespace GitFileChangeNode {
+    export function is(node: any): node is GitFileChangeNode {
+        return 'uri' in node && 'status' in node && 'description' in node && 'label' in node && 'icon' in node;
```

Comment:
`return node && 'uri' in node && etc..`

Reply:
we should use `Object | undefined` instead of `any` to distinguish undefined from everything else

Reply:
I second that, that would be even better but we already use `any` all over the code.

Reply:
...at least for public APIs, i.e. since `SelectionService.selection` is any, ts won't complain that it can be undefined